common4j/src/main/com/microsoft/identity/common/java/cache/SharedPreferencesAccountCredentialCacheWithMemoryCache.java

```diff
@@ -94,6 +97,10 @@ private void load() {
                 Logger.error(methodTag, "Failed to load initial accounts or credentials from SharedPreferences", t);
             } finally {
                 mLoaded = true;
+                if (!mSha1Cleared) {
+                    saveSha1ClearedFlag();
+                    mSha1Cleared = true;
+                }
```

Comment:
Originally I was going to suggest removing the conditional and just saving() every time, but it does force a full I/O write of the whole sharedpreferences to do that, so better to keep this conditional.  With that said, with the advice above, you won't need to update mSha1Cleared anymore afterwards.

I do think this block should only be inside the try.  If we Throw for some reason during the load, I don't expect the write to be any more successful.  And if we're throwing on the save() operation, we have no handler here and likely crash because it will get to the top level of our thread.

Reply:
Going to resolve this comment because it's referring to a different conditional that I had before I did some more refactoring, and that might be confusing to read in the future.